numpy/core/src/multiarray/dragon4.h

```diff
@@ -40,6 +40,28 @@
 #include "npy_pycompat.h"
 #include "numpy/arrayscalars.h"
 
+
+#define NPY_HALF_BINFMT_NAME IEEE_binary16
+
+/* NPY_FLOAT_BINFMT_NAME and NPY_DOUBLE_BINFMT_NAME defined in npy_common.h */
```

Comment:
Yeah, I'll move it.

Reply:
Hmm for reasons I don't entirely understand yet, the macro-variable `HAVE_NPY_CONFIG_H` is false during the build, meaning I cannot access the `HAVE_LDOUBLE*` macros inside `npy_common.h`.

Curiously, if inside `npy_common.h` I delete the line `#ifdef HAVE_NPY_CONFIG_H` and try to import `npy_config.h`, everthing compiles fine...

Reply:
That is strange, it is defined as part of the compiler command line in `numpy/core/setup.py`. Maybe something is out of order?

Reply:
I partly worked it out, but I've also decided to avoid trying to fix this by just moving these defs to `dragon4.h`.

The first problem was that "libraries" like npymath (as opposed to "extensions") do not currently inherit macros from the distutils config object. That is easy to fix by adding a line or two to inherit the macros in `numpy/distutils/misc_util.py` in `_add_library`.

But the next problem I didn't solve is that a number of non-core modules (like fft) also need to import `npy_common.h` too, but since they are not submodules of `core` they cannot inherit the macros. 

Looking in `core/setup.py` there are comments about hacks needed to get `config.h` to be generated at the right time (see "dummymodule") which I am scared of messing with, and its unclear to me why the `HAVE_NPY_CONFIG_H` exists at all, so I'm going to refocus on finishing these dragon4 changes.